### DISCUSSION TITLE
how to feed local computer audio to the liquidsoap in windows

### DISCUSSION THREAD

**Discussion by vannan5** [savonet/liquidsoap#2528] [Q&A]
how to feed local computer audio to the liquidsoap in windows

**Comment by add-n2x** [comment]
You may want to research about "Loopback Audio".

A popular free tool capable of Loopback Audio is [JACK Audio](https://jackaudio.org/faq/jack_on_windows.html) or [VB Cable](https://vb-audio.com/Cable/). Some audio interfaces also support loopback channels natively. Or you route some output to an input by simply patching a cable ; )

As soon you create such "input channel" with one of these options, you should be able to use it in Liquidsoap as an input too.

**Comment by vannan5** [comment]
thank you so much for the reply.as a alternative we are going with BUTT tool for broadcast. if there is too much latency. we will try this solution it and update you.

**Reply by add-n2x** [reply]
Okay, in that case I'm not sure if I understood your initial question clearly. I was assuming your are talking about your computer's audio output.

If you want to use your computer's soundcard audio input then you can get it like this:

```ocaml
input = input.portaudio(id="input_windows")
output.icecast(%vorbis, mount="some.ogg", input)
```

Note this is just a rough, untested script. Add details as per [documentation](https://www.liquidsoap.info/doc-2.1.0).
